backend/api/authentication.go

```diff
@@ -71,11 +71,12 @@ func (a *AuthenticationClient) TestCreds(ctx context.Context, username, password
 	if basic, ok := providers[basic.Type]; ok {
 		if _, err := basic.Authenticate(ctx, username, password); err == nil {
 			return nil
+		} else {
+			return err
```

Comment:
I don't think we should return the actual error to the user (maybe just log it), simply because we would actually reveal whether a user actually exists or not and if it's because the password is wrong, e.g. https://github.com/sensu/sensu-go/blob/572b4b5facc4a4f7b310dcf0eb34ef711e9583e5/backend/store/etcd/user_store.go#L40-L43

Reply:
Let's log the store error, keep it as is, and write a different error to the user in the HTTP handler.

Reply:
Thanks for raising that @palourde; just like @echlebek I think that "obfuscation" belongs to the HTTP handlers, the boundary between the outside and the inside. Internally, we want the most precise error we can get.

Does that work for you @palourde?

Reply:
Ah yes all good, I didn't realized the HTTP handler does not respond with the error!

Reply:
So based on [the current handler code](https://github.com/sensu/sensu-go/blob/master/backend/apid/routers/authentication.go#L70-L89) we agree it's all good? We log the exact error in the backend's logs but we only tell the client it's a 401 status code.